src/components/common/AwardsScroller.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+
+interface AwardsScrollerProps {
+  winnerPlace: string;
+
+  setWinnerPlace: (place: string) => void;
+}
+
+const AwardsScroller: React.FC<AwardsScrollerProps> = ({
+  winnerPlace,
+  setWinnerPlace,
+}) => {
+  const handleScrollToSection = (sectionId: string) => {
+    const section = document.getElementById(sectionId);
+    if (section) {
+      section.scrollIntoView({ behavior: 'smooth' });
+    }
+    setWinnerPlace(sectionId); // Update the selected button
+  };
+
+  if (winnerPlace === 'S') {
+    console.log('Special Awards');
+  }
+
+  return (
+    <div className="flex flex-col overflow-hidden text-ellipsis whitespace-nowrap text-xs">
+      <button
+        onClick={() => handleScrollToSection('1')}
+        className={`text-left ${
+          winnerPlace === '1' ? 'text-white' : 'text-xs text-gray-300'
+        }`}
+      >
+        🏆 1st Place
+      </button>
+      <button
+        onClick={() => handleScrollToSection('2')}
+        className={`text-left ${
+          winnerPlace === '2' ? 'text-white' : 'text-xs text-gray-300'
+        }`}
+      >
+        🥈 2nd Place
+      </button>
+      <button
+        onClick={() => handleScrollToSection('3')}
+        className={`truncate text-left ${
+          winnerPlace === '3' ? 'text-white' : 'text-xs text-gray-300'
+        }`}
+      >
+        🏆 3rd Place
+      </button>
+      <button
+        onClick={() => handleScrollToSection('S')}
+        className={`truncate text-left ${
+          winnerPlace === 'S' ? 'text-white' : 'text-xs text-gray-300'
+        }`}
+      >
+        🌟 Special Awards
+      </button>
+    </div>
+  );
+};
+
+export default AwardsScroller;
```

Comment:
You could do a refactor to map over data instead of hardcoding each individual button, e.g:


```ts
import React from 'react';

interface AwardsScrollerProps {
  winnerPlace: string;
  setWinnerPlace: (place: string) => void;
}

interface Award {
  id: string;
  label: string;
  emoji: string;
}

const awards: Award[] = [
  { id: '1', label: '1st Place', emoji: '🏆' },
  { id: '2', label: '2nd Place', emoji: '🥈' },
  { id: '3', label: '3rd Place', emoji: '🏅' },
  { id: 'S', label: 'Special Awards', emoji: '🌟' },
];

const AwardsScroller: React.FC<AwardsScrollerProps> = ({ winnerPlace, setWinnerPlace }) => {
  const handleScrollToSection = (sectionId: string) => {
    const section = document.getElementById(sectionId);
    if (section) {
      section.scrollIntoView({ behavior: 'smooth' });
    }
    setWinnerPlace(sectionId);
  };

  return (
    <div className="flex flex-col overflow-hidden text-ellipsis whitespace-nowrap text-xs">
      {awards.map(({ id, label, emoji }) => (
        <button
          key={id}
          onClick={() => handleScrollToSection(id)}
          className={`text-left ${
            winnerPlace === id ? 'text-white' : 'text-xs text-gray-300'
          } truncate`}
        >
          {emoji} {label}
        </button>
      ))}
    </div>
  );
};

export default AwardsScroller;
```

Make sure to put the data in the data.ts folder 🫡